src/NexusMods.App.UI/Pages/LoadoutPage/LoadoutViewModel.cs

```diff
@@ -98,6 +128,56 @@ public LoadoutViewModel(IWindowManager windowManager, IServiceProvider servicePr
                 .AddTo(disposables);
         });
     }
+
+    
+    /// <summary>
+    /// Returns the appropriate LoadoutItemGroup of files if the selection contains a LoadoutItemGroup containing files,
+    /// if the selection contains multiple LoadoutItemGroups of files, returns None.
+    /// </summary>
+    private async Task<Optional<LoadoutItemGroup.ReadOnly>> GetViewModFilesLoadoutItemGroup(CancellationToken cancellationToken)
+    {
+        return await Task.Run(() =>
+            {
+                var loadoutItemIds = Adapter.SelectedModels[0].GetLoadoutItemIds();
+                // Only allow when selecting a single item, or an item with a single child
+                if (loadoutItemIds.Count != 1) return Optional<LoadoutItemGroup.ReadOnly>.None;
+
+                var loadoutItem = LoadoutItem.Load(_connection.Db, loadoutItemIds.First());
+
+                if (!loadoutItem.TryGetAsLoadoutItemGroup(out var currentGroup)) return Optional<LoadoutItemGroup.ReadOnly>.None;
+
+                // Allow groups of files, or hierarchy of single child groups that end with a group of files
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    var children = LoadoutItem.FindByParent(_connection.Db, currentGroup.Id);
+
+                    switch (children.Count)
+                    {
+                        case 0:
+                            return Optional<LoadoutItemGroup.ReadOnly>.None;
+                        case 1:
+                        {
+                            var firstChild = children.First();
+
+                            if (!firstChild.TryGetAsLoadoutItemGroup(out var nextGroup))
+                            {
+                                return firstChild.IsLoadoutItemWithTargetPath() ? currentGroup : Optional<LoadoutItemGroup.ReadOnly>.None;
+                            }
+
+                            // check if the single child group is valid
+                            currentGroup = nextGroup;
+                            continue;
+                        }
+                        default:
+                            return children.First().IsLoadoutItemWithTargetPath() ? currentGroup : Optional<LoadoutItemGroup.ReadOnly>.None;
+                    }
+                }
+
+                return Optional<LoadoutItemGroup.ReadOnly>.None;
+            },
+            cancellationToken: cancellationToken
+        );
```

Comment:
This can be replaced with the code used in the `LoadoutDataProviderHelper`. That code is much simpler and faster.